board.cpp

```diff
@@ -40,41 +43,111 @@ QVariant Board::data(const QModelIndex &index, int role) const
         return m_data[index.row()/m_dimension][index.row()%m_dimension].isActive();
     case Qt::UserRole + 1:
         return m_data[index.row()/m_dimension][index.row()%m_dimension].pos();
+    case Qt::UserRole + 2:
+        return m_data[index.row()/m_dimension][index.row()%m_dimension].selected();
     default:
         break;
     }
-
     return QVariant();
 }
 
 QHash<int, QByteArray> Board::roleNames() const
 {
-    return { {Qt::DecorationRole, "color"}, {Qt::UserRole, "active"}, {Qt::UserRole + 1, "index"} };
+    return { {Qt::DecorationRole, "color"}, {Qt::UserRole, "active"}, {Qt::UserRole + 1, "index"}, {Qt::UserRole + 2, "selected"} };
```

Comment:
`{ 
  {Qt::DecorationRole, "color"},
  {Qt::UserRole, "active"},
  {Qt::UserRole + 1, "index"},
  {Qt::UserRole + 2, "selected"} 
};`

looks better

Reply:
Also good approach is to add private enum
`enum RoleNames {
       ColorRole = Qt::DecorationRole,
       ACtiveRole = Qt::UserRole,
       IndexRole,
       SelectedRole
}`

and after that you use enum values intead of Qt roles